src/mono/mono/mini/mini-amd64.c

```diff
@@ -1077,6 +1076,21 @@ get_call_info (MonoMemPool *mp, MonoMethodSignature *sig)
 		default:
 			g_assert_not_reached ();
 		}
+
+#ifdef MONO_ARCH_HAVE_SWIFTCALL
+		if (mono_method_signature_has_ext_callconv (sig, MONO_EXT_CALLCONV_SWIFTCALL)) {
+			MonoClass *swift_error_ptr = mono_class_try_get_swift_error_ptr_class ();
+			MonoClass *swift_self = mono_class_try_get_swift_self_class ();
+			MonoClass *klass = mono_class_from_mono_type_internal (sig->params [i]);
+			if (klass) {
```

Comment:
Aren't we overwritting the previously assigned arg storage ? So, this param would first be handled as a `MONO_TYPE_I` with `add_general`, potentially reserving stack space to pass it if all the param registers are consumed ? I'm not sure if this is correct.

Reply:
I think you are correct. Currently, we treat SwiftSelf as ArgVtypeInIRegs and SwiftError as ArgInIReg. There might be cases where they should be passed via the stack. I see two possible options:
 - Add a flag to ArgInfo
 - Always pass them as stack parameters

Reply:
I thought the swift cconv states that they are always passed in regs. ?

Reply:
After offline sync, we determined that the new argument types in ArgInfo lack information on arg storage (param regs or stack). The code is updated to directly use context registers during the call_emit function.